query/src/interpreters/interpreter_insert_into.rs

```diff
@@ -64,6 +72,39 @@ impl Interpreter for InsertIntoInterpreter {
                 ValueSource::new(Cursor::new(values), self.plan.schema(), block_size);
             let stream_source = SourceStream::new(Box::new(values_source));
             stream_source.execute().await
+        } else if let Some(select_executor) = &self.select {
+            let output_schema = self.plan.schema();
+            let select_schema = select_executor.schema();
+            if select_schema.fields().len() < output_schema.fields().len() {
+                return Err(ErrorCode::BadArguments("DataBlock schema mismatch"));
+            }
+
+            let select_input_stream = select_executor.execute(None).await?;
+            let cast_input_stream = select_input_stream.map(move |data_block| match data_block {
+                Err(fail) => Err(fail),
+                Ok(data_block) => {
+                    let rows = data_block.num_rows();
+                    let iter = output_schema.fields().iter().zip(select_schema.fields());
+                    let mut colunm_vec = vec![];
+                    for (i, (output_field, input_field)) in iter.enumerate() {
+                        let func = cast_function(output_field.data_type())?;
```

Comment:
we can keep  `funcs`  out of the stream iter, because trait Function is Send + Sync.